python/phylanx/util/__init__.py

```diff
@@ -56,6 +77,39 @@ def recompile(self,a,allowreturn=False):
             else:
                 raise Exception()
             return s
+        elif nm == "Subscript":
+            e = get_node(a,name="ExtSlice")
+            s0 = get_node(e,name="Slice",num=0)
+            s1 = get_node(e,name="Slice",num=1)
+            xlo = s0.lower
+            xhi = s0.upper
+            ylo = s1.lower
+            yhi = s1.upper
+            sname = self.recompile(get_node(a,num=0))
+            s = "slice("
+            s += sname
+            s += ","
+            if xlo == None:
+                s += "0"
+            else:
+                s += self.recompile(xlo)
+            s += ","
+            if xhi == None:
+                s += "shape(" + sname + ",0)"
+            else:
+                s += self.recompile(xhi)
+            s += ","
+            if ylo == None:
+                s += "0"
+            else:
+                s += self.recompile(ylo)
+            s += ","
+            if yhi == None:
+                s += "shape(" + sname + ",1)"
```

Comment:
You could use `-1` as the index to refer to the last column.

Reply:
Actually, seeing this comment makes me realize the semantics of slice() does not match Python. The -1 in the last position is not supposed to work. In addition, it should be possible to have a third argument.
```
>>> r = range(10)
>>> print r[:]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> print r[0:-1]
[0, 1, 2, 3, 4, 5, 6, 7, 8]
>>> print r[::2]
[0, 2, 4, 6, 8]
>>> print r[1::2]
[1, 3, 5, 7, 9]
>>> print r[::-1]
[9, 8, 7, 6, 5, 4, 3, 2, 1, 0]
````

Reply:
Yes, you're certainly right.